### ISSUE TITLE
make fails on apple M1/arm64 with "error: addend too big for relocation"

### ISSUE THREAD

**Issue by StevePny** [NOAA-GFDL/FRE-NCtools#147]
**Describe the bug**
During make:

```
Making all in tools/simple_hydrog/lakes
gfortran -I../../../tools/simple_hydrog/libfmslite -fdefault-real-8 -I/opt/homebrew/Cellar/netcdf/4.8.1/include -I/opt/homebrew/Cellar/netcdf/4.8.1/include -g -O2 -ffree-line-length-none -fcray-pointer -c -o cr_lake_files.o  cr_lake_files.f90
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:1792:2: error: addend too big for relocation
        adrp    x5, _idat.3@PAGE+16577280
        ^
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:1796:2: error: addend too big for relocation
        adrp    x4, _idat.3@PAGE+16583040
        ^
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:1799:2: error: addend too big for relocation
        add     x5, x5, _idat.3@PAGEOFF+16577280;momd
        ^
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:1799:2: error: fixup value out of range
        add     x5, x5, _idat.3@PAGEOFF+16577280;momd
        ^
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:1800:2: error: addend too big for relocation
        add     x4, x4, _idat.3@PAGEOFF+16583040;momd
        ^
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:1800:2: error: fixup value out of range
        add     x4, x4, _idat.3@PAGEOFF+16583040;momd
        ^
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:4699:2: error: addend too big for relocation
        adrp    x0, _wbdat.5@PAGE+33177600
        ^
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:4700:2: error: addend too big for relocation
        add     x0, x0, _wbdat.5@PAGEOFF+33177600;momd
        ^
/var/folders/z8/mbk72w1n74343wphmqqv21h80000gn/T//ccoXwgje.s:4700:2: error: fixup value out of range
        add     x0, x0, _wbdat.5@PAGEOFF+33177600;momd

```

**To Reproduce**
make on apple M1/arm64 platform

**Expected behavior**
Successful build

**System Environment**
Describe the system environment, include:
- OS: MacOSX Monterey 12.1
- Compiler(s): Type and version [e.g. Intel 19.1]
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 13.0.0 (clang-1300.0.29.30)
Target: arm64-apple-darwin21.2.0
Thread model: posix
GNU Fortran (Homebrew GCC 11.2.0_3) 11.2.0

- netCDF Version: For both C and Fortran
netCDF 4.8.1

- Configure options: n/a

**Additional context**
This seems to be related to the gcc version currently used on Mac M1
https://github.com/Homebrew/discussions/discussions/2687
https://windowsquestions.com/2021/05/24/error-while-compiling-a-simple-c-program-with-g-11-on-m1-macbook/

**Comment by underwoo** [comment]
I am able to reproduce this on my M1 Mac, with all the latest software (as of 23 Feb 2022).  Doing some searching, it may be the gfortran/clang for ARM may not be fully baked.  We will need to investigate this more.  However, the number of us that have access to M1 Mac or other ARM hardware, it may take us a while to resolve this.

**Comment by ngs333** [comment]
@StevePny 
I have tried compiling on ARM hardware with several compilers (including gcc 11) and have not been able to reproduced your issue. In the process I have learned of another problem (see https://github.com/NOAA-GFDL/FRE-NCtools/issues/174 ) that should be fixed shortly, but its not likely to be related to yours. 

One thing you might try is to increase the stack size available to your programs (including the compiler).  Try ```ulimit -a``` to see what is is, then perhaps ```ulimit -s 65536``` to increase it to 65MB.
